superx/information_extractors/item_info_extractor.py

```diff
@@ -0,0 +1,248 @@
+                                                       'item_date_format': '%Y-%m-%d'}
+                                           }
+
+    def run_info_extractor(self):
+        """
+        This method starts the extraction process
+        """
+        for key in self.super_specific_information:
+            self.current_super = self.super_specific_information[key]
+            url_list = [self.current_super['url']]
+
+            if self.current_super['multiple_pages']:
+                try:
+                    url_list = self.get_all_super_links()
+                except ConnectionError as ce:
+                    logging.error(str(ce))
+
+            zip_links = self.get_zip_file_links(url_list)
+            self.info_parser(zip_links)
+
+    def get_zip_file_links(self, url_list):
+        """
+        This method web scrapes the urls in url_list and creates a set of the gzip file links.
+        The method then sends the set to parsing
+        If connection to the url failed, moves on to next url
+        :param url_list: list of urls to extract zip files from
+
+        """
+        for url in url_list:
+            try:
+                page = requests.get(url)
+                web_scrapper = BeautifulSoup(page.content, 'html.parser')
+            except Exception:
+                logging.error(f'Unable to connect to url:\n{url}')
+            else:
+                links_list = web_scrapper.find_all('a')
+                zip_links = set()
+
+                for link in links_list:
+                    if link.has_attr('href'):
+                        https = str(link.attrs['href'])
+                        if self.current_super['price_full'] in https:
+                            zip_links.add(https)
+
+                return zip_links
+
+    def info_parser(self, zip_links):
+        """
+        This method retrieves the xml file in the gzip file and parses it into an xml tree
+        The branch_id is retrieved from the xml file and then it is sent for retrieval of the item information
+        If connection failed, moves on to next link
+        :param zip_links: list of zip file links from the website
+        """
+        for zip_link in zip_links:
+            # fix zip link url if neccessary
+            if not self.current_super['zip_link_prefix'] is None:
+                zip_link = self.current_super['zip_link_prefix'] + zip_link
+
+            try:
+                request = requests.get(zip_link)
+                content = request.content
+            except Exception:
+                logging.error(f'Unable to extract from zip file with url: {zip_link}')
+            else:
+                xml_file = gzip.decompress(content).decode('utf-8')
+                # parses the xml document into a tree
+                tree = ET.fromstring(xml_file)
+                # gets child containing item information
+                items = tree.getchildren()[-1]
+                self.extract_information(items)
```

Comment:
@simzacks question about this. instead of calling the next method at the end of this method, should i instead creat a list of item nodes, return that list and call extract _info with the list (in a for loop obviously). The reasoning behind is to make the script more testable.

Reply:
I don't see why this would be more testable. You jcan build a test XML file, or just a set of nodes that have the relevant fields and pass that into the function.